### PR TITLE
"Unrelated multi-clause function" added (Anti-patterns documentation)

### PR DESCRIPTION
**Pull request by lucasvegi**

### CODE REVIEW

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 1]
```diff
@@ -132,7 +132,84 @@ iex> AlternativeInteger.parse_discard_rest("13")
...
 ## Unrelated multi-clause function
 
-TODO
+#### Problem
+
+Using multi-clause functions in Elixir, to group functions of the same name, is not an anti-pattern in itself. However, due to the great flexibility provided by this programming feature, some developers may abuse the number of guard clauses and pattern matches to group *unrelated* functionality.
+
+#### Example
+
+A recurrent example of abusive use of multi-clause functions is when we’re trying to mix too much-unrelated business logic into the function definitions. This makes it difficult to read and understand the logic involved in the functions, which may impair code maintainability. Some developers use documentation mechanisms such as `@doc` annotations to compensate for poor code readability, but unfortunately, with a multi-clause function, we can only use these annotations once per function name, particularly on the first or header function. As shown next, all other variations of the function need to be documented only with comments, a mechanism that cannot automate tests, leaving the code prone to bugs.
```
Comment (whatyouhide) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:141]:
```suggestion
A frequent example of this usage of multi-clause functions is when developers mix unrelated business logic into the function definitions. This makes it difficult to read and understand the logic involved in the functions, which may impair code maintainability. Some developers use documentation mechanisms such as `@doc` annotations to compensate for poor code readability, but unfortunately, with a multi-clause function, we can only use these annotations once per function name, particularly on the first or header function. As shown next, all other variations of the function need to be documented only with comments, a mechanism that cannot automate tests, leaving the code prone to bugs.
```

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 2]
```diff
@@ -132,7 +132,84 @@ iex> AlternativeInteger.parse_discard_rest("13")
...
+
+```elixir
+@doc """
+Update sharp product with 0 or empty count
+
+## Examples
+iex> Namespace.Module.update(...)
+expected result...
```
Comment (whatyouhide) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:149]:
```suggestion
## Examples

    iex> Namespace.Module.update(...)
    ...

```

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 3]
```diff
@@ -132,7 +132,84 @@ iex> AlternativeInteger.parse_discard_rest("13")
...
+
+Using multi-clause functions in Elixir, to group functions of the same name, is not an anti-pattern in itself. However, due to the great flexibility provided by this programming feature, some developers may abuse the number of guard clauses and pattern matches to group *unrelated* functionality.
+
+#### Example
+
+A recurrent example of abusive use of multi-clause functions is when we’re trying to mix too much-unrelated business logic into the function definitions. This makes it difficult to read and understand the logic involved in the functions, which may impair code maintainability. Some developers use documentation mechanisms such as `@doc` annotations to compensate for poor code readability, but unfortunately, with a multi-clause function, we can only use these annotations once per function name, particularly on the first or header function. As shown next, all other variations of the function need to be documented only with comments, a mechanism that cannot automate tests, leaving the code prone to bugs.
+
+```elixir
+@doc """
+Update sharp product with 0 or empty count
```
Comment (whatyouhide) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:145]:
```suggestion
Updates "sharp" product with `0` or empty count.
```

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 4]
```diff
@@ -132,7 +132,84 @@ iex> AlternativeInteger.parse_discard_rest("13")
...
+
+## Examples
+iex> Namespace.Module.update(...)
+expected result...
+"""
+def update(%Product{count: nil, material: material}) when material in ["metal", "glass"] do
+  # ...
+end
+
+# update blunt product
```
Comment (whatyouhide) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:155]:
```suggestion
# Updates "blunt" product
```

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 5]
```diff
@@ -132,7 +132,84 @@ iex> AlternativeInteger.parse_discard_rest("13")
...
+def update(%Product{count: count, material: material}) when count > 0 and material in ["metal", "glass"] do
+  # ...
+end
+
+# update animal...
```
Comment (whatyouhide) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:160]:
```suggestion
# Updates animal
```

File: lib/elixir/pages/anti-patterns/design-anti-patterns.md [hunk 6]
```diff
@@ -132,7 +132,84 @@ iex> AlternativeInteger.parse_discard_rest("13")
...
+## Parameter
+struct: %Animal{...}
+
+## Examples
+iex> Namespace.Module.update_animal(%Animal{...})
+expected result...
+"""
+def update_animal(struct) do
+  # ...
+end
```
Comment (whatyouhide) [inline lib/elixir/pages/anti-patterns/design-anti-patterns.md:211]:
```suggestion
@doc """
Updates a "sharp" product.

## Examples

    iex> Namespace.Module.update_sharp_product(%Product{...})
    ...

"""
@spec update_sharp_product(Product.t()) :: term()
def update_sharp_product(struct) do
  # ...
end

@doc """
Updates a "blunt" product.

## Examples

    iex> Namespace.Module.update_blunt_product(%Product{...})
    ...
    
"""
@spec update_blunt_product(Product.t()) :: term()
def update_blunt_product(struct) do
  # ...
end

@doc """
Updates an animal.

## Examples

    iex> Namespace.Module.update_animal(%Animal{...})
    ...

"""
@spec update_animal(Animal.t()) :: term()
def update_animal(struct) do
  # ...
end
```

**Comment by lucasvegi** [comment]
👍

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**